src/rf/apps/workers/status_updates.py

```diff
@@ -10,49 +10,234 @@
+        layer.status_thumbnail_start = (value if layer.status_thumbnail_start
+                                        is None else layer.status_upload_start)
+
+    elif status == enums.STATUS_CREATE_CLUSTER:
+        layer.status_create_cluster_start = (value if
+                                             layer.status_create_cluster_start
+                                             is None else
+                                             layer.status_upload_start)
+
+    elif status == enums.STATUS_CHUNK:
+        layer.status_chunk_start = (value if layer.status_upload_start
+                                    is None else layer.status_upload_start)
+
+    elif status == enums.STATUS_MOSAIC:
+        layer.status_mosaic_start = (value if layer.status_upload_start
+                                     is None else layer.status_upload_start)
+
+    return layer
+
+
+def update_layer_status_end(layer, status, error_message=None):
+    value = datetime.now()
+    if status == enums.STATUS_UPLOAD:
+        layer.status_upload_end = value
+        layer.status_upload_error = error_message
+    elif status == enums.STATUS_VALIDATE:
+        layer.status_validate_end = value
+    elif status == enums.STATUS_THUMBNAIL:
+        layer.status_thumbnail_end = value
+        layer.status_thumbnail_error = error_message
+    elif status == enums.STATUS_CREATE_CLUSTER:
+        layer.status_create_cluster_end = value
+        layer.status_create_cluster_error = error_message
+    elif status == enums.STATUS_CHUNK:
+        layer.status_chunk_end = value
+        layer.status_chunk_error = error_message
+    elif status == enums.STATUS_MOSAIC:
+        layer.status_mosaic_end = value
+        layer.status_mosaic_error = error_message
+    elif status == enums.STATUS_COMPLETED:
+        layer.status_completed = value
+    elif status == enums.STATUS_FAILED:
+        layer.status_failed_error = error_message
+
+    # If we had any error message mark the generic failed field.
+    if error_message is not None:
+        layer.status_failed = value
+
+    return layer
+
+
+def update_layerimage_status_start(image, status):
+    value = datetime.now()
+    if status == enums.STATUS_UPLOAD:
+        image.status_upload_start = value
+    elif status == enums.STATUS_VALIDATE:
+        image.status_validate_start = value
+    elif status == enums.STATUS_THUMBNAIL:
+        image.status_thumbnail_start = value
+    return image
+
+
+def update_layerimage_status_end(image, status, error_message=None):
+    value = datetime.now()
+    if status == enums.STATUS_UPLOAD:
+        image.status_upload_end = value
+        image.status_upload_error = error_message
+    elif status == enums.STATUS_VALIDATE:
+        image.status_validate_end = value
+        image.status_validate_error = error_message
+    elif status == enums.STATUS_THUMBNAIL:
+        image.status_thumbnail_end = value
+        image.status_thumbnail_error = error_message
+    return image
```

Comment:
We could drop these enums and make things easier to read by converting these to methods on the model class. Consider:

  * `image.upload_started()` instead of `update_layerimage_status_start(image, enums.STATUS_UPLOAD`
  * `image.upload_ended('blah')` instead of `update_layerimage_status_end(image, enums.STATUS_UPLOAD, 'blah')`
  * etc.

Same for the layer helper functions.